### PR TITLE
feat: Promote WorkstationConfig / Workstation to v1beta1

### PR DESCRIPTION
**Pull request by jasonvigil**
Commit 1: Add fuzzer for WorkstationConfig
Commit 2: Add fuzzer for Workstation
Commit 3: Promote WorkstationConfig to v1beta1
Commit 4: Promote Workstation to v1beta1

### CODE REVIEW

File: pkg/test/resourcefixture/testdata/basic/workstations/workstationconfig/workstationconfig-full/_generated_object_workstationconfig-full.golden.yaml [hunk 1]
```diff
@@ -1,6 +1,8 @@
-apiVersion: workstations.cnrm.cloud.google.com/v1alpha1
+apiVersion: workstations.cnrm.cloud.google.com/v1beta1
 kind: WorkstationConfig
 metadata:
+  annotations:
+    cnrm.cloud.google.com/management-conflict-prevention-policy: none
```
Comment (yuwenma) [inline pkg/test/resourcefixture/testdata/basic/workstations/workstationconfig/workstationconfig-full/_generated_object_workstationconfig-full.golden.yaml:5]: Do you know why this annotation is applied only in beta?

Comment (jasonvigil) [reply pkg/test/resourcefixture/testdata/basic/workstations/workstationconfig/workstationconfig-full/_generated_object_workstationconfig-full.golden.yaml:5]: I think this is a bug. The management conflict annotation defaulter webhook is incorrectly applying the annotation for direct resources. Fixed in https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/3417

**Comment by jasonvigil** [comment]
split into:
* https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/3410
* https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/3411
* https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/3412
* https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/3430

**Pull request closed without merging**